app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -120,73 +123,78 @@ def process_collection(collection, key)
         cloud_subnets = get_cloud_subnets(cloud_network).collect do |raw_subnet|
-          @data_index.fetch_path(:cloud_subnets, raw_subnet.fetch_path('id'))
+          @data_index.fetch_path(:cloud_subnets, raw_subnet.id)
         end
 
-        uid = cloud_network.fetch_path('id')
+        uid = resource_uid(@subscription_id,
+                           cloud_network.resource_group.downcase,
+                           cloud_network.type.downcase,
+                           cloud_network.name)
+
         new_result = {
-          :ems_ref       => uid,
-          :name          => cloud_network.fetch_path('name'),
-          :cidr          => cloud_network.fetch_path('properties', 'addressSpace', 'addressPrefixes').join(", "),
-          :enabled       => true,
-          :cloud_subnets => cloud_subnets,
+          :ems_ref             => cloud_network.id,
+          :name                => cloud_network.name,
+          :cidr                => cloud_network.properties.address_space.address_prefixes.join(", "),
+          :enabled             => true,
+          :cloud_subnets       => cloud_subnets,
+          :orchestration_stack => @data_index.fetch_path(:orchestration_stacks, @resource_to_stack[uid]),
         }
         return uid, new_result
       end
 
       def parse_cloud_subnet(subnet)
-        uid = subnet.fetch_path('id')
+        uid = subnet.id
         new_result = {
           :ems_ref => uid,
-          :name    => subnet.fetch_path('name'),
-          :cidr    => subnet.fetch_path('properties', 'addressPrefix'),
+          :name    => subnet.name,
+          :cidr    => subnet.properties.address_prefix,
         }
         return uid, new_result
       end
 
       def parse_instance(instance)
         uid = resource_uid(@subscription_id,
-                           instance.fetch_path('resourceGroup'),
-                           instance.fetch_path('type').downcase,
-                           instance.fetch_path('name'))
-        series_name = instance.fetch_path('properties', 'hardwareProfile', 'vmSize')
-        az          = instance.fetch_path('properties', 'availabilitySet', 'id')
+                           instance.resource_group.downcase,
+                           instance.type.downcase,
+                           instance.name)
+        series_name = instance.properties.hardware_profile.vm_size
+        az          = instance.properties.try(:availability_set).try(:id)
         series      = @data_index.fetch_path(:flavors, series_name)
 
         new_result = {
           :type                => 'ManageIQ::Providers::Azure::CloudManager::Vm',
           :uid_ems             => uid,
           :ems_ref             => uid,
-          :name                => instance.fetch_path('name'),
+          :name                => instance.name,
           :vendor              => "Microsoft",
-          :raw_power_state     => instance["powerStatus"],
+          :raw_power_state     => power_status(instance),
```

Comment:
armrest gem no longer automatically fetches power status. We need to explicitly make an API request to get the status. See the implementation of `power_status` method.